tests/conftest.py

```diff
@@ -83,3 +110,131 @@ def wait_for_container(
     else:
         log.error("Log from container: %s", container.logs().decode())
         raise TimeoutError("Timeout while waiting for model loading")
+
+
+def is_capella_5_x_x() -> bool:
+    return bool(re.match(r"5.[0-9]+.[0-9]+", os.getenv("CAPELLA_VERSION", "")))
+
+
+def is_capella_6_x_x() -> bool:
+    return bool(re.match(r"6.[0-9]+.[0-9]+", os.getenv("CAPELLA_VERSION", "")))
+
+
+def get_projects_of_t4c_repository(t4c_http_port: str) -> list[dict[str, str]]:
+    res = requests.get(
+        f"http://localhost:{int(t4c_http_port)}/api/v1.0/projects/{T4C_REPO_NAME}",
+        auth=_get_basic_auth(),
+        timeout=60,
+    )
+    res.raise_for_status()
+    return res.json()["projects"]
+
+
+# In case you encounter a connection error here for the tests, first try:
+# 1. deamon-reload
+# 2. docker restart
+# Related to: https://github.com/testcontainers/testcontainers-go/issues/551
+def create_t4c_repository(t4c_http_port: str):
+    res = requests.post(
+        f"http://localhost:{t4c_http_port}/api/v1.0/repositories",
+        auth=_get_basic_auth(),
+        timeout=60,
+        json={
+            "repositoryName": T4C_REPO_NAME,
+            "authenticationType": "",
+            "datasourceType": "H2_EMBEDDED",
+        },
+    )
+    res.raise_for_status()
+
+    assert res.status_code == 201
+
+
+def _get_basic_auth() -> auth.HTTPBasicAuth:
+    return auth.HTTPBasicAuth(HTTP_LOGIN, HTTP_PASSWORD)
+
+
+@pytest.fixture(name="local_git_container")
+def fixture_local_git_container() -> containers.Container:
+    ports: dict[str, int | None] = {"80/tcp": None}
+
+    with get_container(image="local-git-server", ports=ports) as container:
+        wait_for_container(container, "server started")
+        yield container
+
+
+@pytest.fixture(name="git_port")
+def fixture_git_port(local_git_container: containers.Containe) -> str:
+    local_git_container.reload()
+    return local_git_container.ports["80/tcp"][0]["HostPort"]
+
+
+@pytest.fixture(name="git_general_environment")
+def fixture_git_general_environment(git_port: str) -> dict[str, str]:
+    return {
+        "FILE_HANDLER": "GIT",
+        "GIT_REPO_URL": f"http://localhost:{git_port}/git/git-test-repo.git",
+        "GIT_REPO_BRANCH": GIT_REPO_BRANCH,
+        "GIT_USERNAME": GIT_USERNAME,
+        "GIT_PASSWORD": GIT_PASSWORD,
+    }
+
+
+@pytest.fixture(name="t4c_server_container")
+def fixture_t4c_server_container() -> containers.Container:
+    env: dict[str, str] = {"REST_API_PASSWORD": "password"}
+
+    ports: dict[str, int | None] = {
```

Comment:
Yes, I could change it but this would also mean that I have to change the `get_container` port type to `dict[str | None]` so we wouldn't have the possibility to provide a concrete port to the `get_container` function.

Reply:
Are you sure? `dict[str | None]` is a subset of `dict[str, int | None]`

Reply:
Yes and it also makes sense for me. In the function you could assign `int` or `None` as value but when you provide an `dict` only having `None` as valid value then there is a problem (since from the functions perspective you could use an `int` as value).
But since we have removed the ports anyway, we can close this.